packages/playground/wordpress/src/index.ts

```diff
@@ -529,20 +531,60 @@ export async function unzipWordPress(php: PHP, wpZip: File) {
 		}
 	}
 
-	if (
-		php.isDir(php.documentRoot) &&
-		isCleanDirContainingSiteMetadata(php.documentRoot, php)
-	) {
-		// We cannot mv the directory over a non-empty directory,
-		// but we can move the children one by one.
-		for (const file of php.listFiles(wpPath)) {
-			const sourcePath = joinPaths(wpPath, file);
-			const targetPath = joinPaths(php.documentRoot, file);
-			php.mv(sourcePath, targetPath);
+	const moveRecursively = (source: string, target: string, php: PHP) => {
+		if (php.fileExists(target)) {
+			/*
+			 * Something exists at the target path.
+			 * Let's check to make sure we aren't copying conflicting types.
+			 *
+			 * In this context, if the source path points to a file and the
+			 * target path points to a directory, we do not intend for the file
+			 * to be moved into the directory.
+			 */
+
+			if (!php.isDir(source) && php.isDir(target)) {
+				throw new Error(
+					`The target ${target} is a directory but the source ${source} is not. This is not supported.`
+				);
+			}
+			if (php.isDir(source) && !php.isDir(target)) {
+				throw new Error(
+					`The source ${source} is a directory but the target ${target} is not. This is not supported.`
+				);
+			}
+		}
+
+		if (isNonEmptyDir(target, php)) {
+			// We cannot move a directory over a non-empty directory,
+			// so we move the children one by one.
+			for (const file of php.listFiles(source)) {
+				const sourcePath = joinPaths(source, file);
+				const targetPath = joinPaths(target, file);
+				moveRecursively(sourcePath, targetPath, php);
+			}
+		} else {
+			if (php.fileExists(target) && !php.isDir(target)) {
+				// Refuse to overwrite existing files to avoid the chance of data loss.
+				const wpPath = source.replace(
+					/^\/tmp\/unzipped-wordpress\//,
+					'/'
+				);
+				logger.warn(
+					`Skipping ${wpPath} because a file exists at the target path.`
+				);
+				return;
+			}
```

Comment:
I'm concerned that we might overwrite existing files. It is an `mv`-style operation, but we copy recursively if moving between different filesystems.

So we are skipping files that already exist.